firebase-common/src/main/java/com/google/firebase/internal/DataCollectionConfigStorage.java

```diff
@@ -61,19 +61,28 @@ public boolean isEnabled() {
     return dataCollectionDefaultEnabled.get();
   }
 
-  public void setEnabled(boolean enabled) {
-    if (dataCollectionDefaultEnabled.compareAndSet(!enabled, enabled)) {
-      sharedPreferences.edit().putBoolean(DATA_COLLECTION_DEFAULT_ENABLED, enabled).apply();
-
-      publisher.publish(
-          new Event<>(DataCollectionDefaultChange.class, new DataCollectionDefaultChange(enabled)));
+  public void setEnabled(Boolean enabled) {
+    if (enabled == null) {
+      sharedPreferences.edit().remove(DATA_COLLECTION_DEFAULT_ENABLED).apply();
+      boolean manifestSetting = readManifestDataCollectionEnabled();
+      if (dataCollectionDefaultEnabled.compareAndSet(!manifestSetting, manifestSetting)) {
+        publisher.publish(
+            new Event<>(
+                DataCollectionDefaultChange.class,
+                new DataCollectionDefaultChange(manifestSetting)));
+      }
+    } else {
+      boolean apiSetting = Boolean.TRUE.equals(enabled);
```

Comment:
Synchronized block?

Reply:
I think shared preferences itself makes sure that it's synchornized

Reply:
I mean Lines 69-73 and lines 77-82 seem extractable into a reusable function

Reply:
yeah makes sense i will do that